src/tests/multihost/alltests/test_krb_access_provider.py

```diff
@@ -236,28 +227,24 @@ def test_0004_k5login_user4_with_deleted_user3(multihost):
         for command in echo_cmds:
             cmd1 = multihost.client[0].run_command(command, raiseonerr=False)
 
-        user = 'foo3'
-        ssh_output = client.auth_from_client(user, 'Secret123')
+        check_login_client(multihost, "foo3", "Secret123")
 
         ldap_cmd = f'ldapdelete -x -H ldap://{multihost.master[0].sys_hostname} \
                    -D "cn=Directory Manager" -w "Secret123" uid=foo3,ou=People,dc=example,dc=test'
-
         cmd2 = multihost.client[0].run_command(ldap_cmd, raiseonerr=False)
-
+        time.sleep(10)
         client_hostname = multihost.client[0].sys_hostname
-        ssh = pexpect_ssh(client_hostname, 'foo4', 'Secret123', debug=False)
-        ssh.login(login_timeout=10, sync_multiplier=1, auto_prompt_reset=False)
-        ssh.command("id > /tmp/accessProvider_id_krb5_004.out 2>&1 ")
-        ssh.command(f"ssh -o StrictHostKeyChecking=no -o PasswordAuthentication=no foo3@{client_hostname}\
-                    'id > /tmp/accessProvider_id_krb5_004.out 2>&1' ")
-        ssh.logout()
-
+        ssh = SSHClient(client_hostname, "foo4", "Secret123")
+        ssh.connect()
+        ssh.execute_command("id > /tmp/accessProvider_id_krb5_004.out 2>&1")
+        ssh.execute_command(f"ssh -o StrictHostKeyChecking=no -o "
+                            f"PasswordAuthentication=no foo3@{client_hostname} id >> "
+                            f"/tmp/accessProvider_id_krb5_004.out 2>&1")
+        ssh.close()
         logfile = multihost.client[0].get_file_contents("/tmp/accessProvider_id_krb5_004.out").decode('utf-8')
-
         multihost.client[0].run_command("truncate -s 0 /home/foo3/.k5login", raiseonerr=False)
```

Comment:
This cleanup will not happen if  ssh.connect() fails with an exception.

Reply:
here  if ssh.connect() fails , there will be nothing to clean up . File will be created after login